File: python/nt_toolbox/read_mesh.py
import numpy as np

def read_mesh(name):
    '''
        reading from a OFF file in 3 dimensions, returning X0 (coordinates) and F (faces)
    '''
    
    with open(name, mode="r") as file:
        
        #check type of file
        file_type = file.readline().strip()
        if file_type != "OFF":
            raise Exception("Wrong type of file, only reads OFF files")
    
        #number of vertices/faces/edges:
        n_verts, n_faces, n_edges = tuple([int(s) for s in file.readline().strip().split(' ')])
        
        #vertices
        X0 = []
        for _ in range(n_verts):
            X0.append(file.readline().strip().split(' '))
            
        #faces
        F = []
        for i in range(n_faces):
            F.append(file.readline().strip().split(' ')[1:])
    
    return np.transpose(np.asarray(X0).astype(float)), np.transpose(np.asarray(F).astype(int))

File: julia/NtToolBox/src/read_mesh.jl
function read_mesh(name)
    """
        reading from a OFF file in 3 dimensions, returning X0 (coordinates) and F (faces)
    """
    file = open(name)

    #check type of file
    file_type = strip(readline(file))
    if file_type != "OFF"
        throw(ArgumentError("Wrong type of file, only reads OFF files"))
    end

    #number of vertices/faces/edges:
    (n_verts, n_faces, n_edges) = tuple([parse(Int,s) for s in split(strip(readline(file)),' ')]...)

    #vertices
    X0 = zeros(3,n_verts)
    for i in 1:n_verts
        X0[:,i]= [parse(Float64,s) for s in split(strip(readline(file)),' ')]
    end

    #faces
    F = zeros(Int,3,n_faces)
    for i in 1:n_faces
        F[:,i] = [parse(Int,s) for s in split(strip(readline(file)),' ')[2:end]]
    end

    return X0, F+1
end
